### DISCUSSION TITLE
Requesting an Input Preset with Noise Reduction and Gate

### DISCUSSION THREAD

**Discussion by duarm** [wwmm/easyeffects#2981] [Q&A] [answered]
Easyeffects doesn't work on my pc because of libadwaita, every libadwaita app on my pc renders context menus black

![image](https://github.com/wwmm/easyeffects/assets/26150581/2edeaf1f-d1dd-4680-986e-990210915bd7)

If someone can make a preset With Noise Reduction and Gate for me, it would help a lot, I tried looking for a preset collection, but there's only output presets, and I couln't find any documentation for each preset.json entry

**Comment by violetmage** [comment]
I have so many questions...

Do you not want to fix your libadwaita problem?

Are you using the flatpak?

Wayland or X11?

**Reply by duarm** [reply]
I want a preset, just that, It's a libadwaita problem, I'm not asking for help with that since this is easyeffects, I'm just asking for a file of an input preset with Noise Reduction and Gate, since I can't make my own because of the libadwaita problem I brought up

Not flatpak, arch package. X11

**Reply by wwmm** [reply, marked as answer]
There you go
[remove_noise.json](https://github.com/wwmm/easyeffects/files/14626197/remove_noise.json). After copying it to `~/.config/easyeffects/input` you can use the command line to load it. Just have in mind that I just added the plugins to the pipeline and saved a file. I did not do any apecial tweak to the settings.

That being said the bug you described is far from normal and may not be a libadwaita bug but something deeper. Maybe in the gpu drivers.

**Reply by duarm** [reply]
Thank you, It happens even with the libadwaita demos, so for sure something libadwaita does is not working, I'm researching

**Reply by wwmm** [reply]
Does anything suspicious appear in the output of `sudo journalctl -fb` after starting a libadwaita app? Does the gtk4 demo also show the problem?

**Reply by duarm** [reply]
same problem on gtk4-demos, the logs only show this

```
(gtk4-demo:7212): Gtk-CRITICAL **: 12:28:01.510: Unable to register the application: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.atspi.Registry was not provided by any .service files

(gtk4-demo:7212): Gtk-CRITICAL **: 12:28:02.702: Unable to register the application: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.atspi.Registry was not provided by any .service files

(gtk4-demo:7212): Gtk-CRITICAL **: 12:28:02.892: Unable to register the application: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.atspi.Registry was not provided by any .service files
```

as far as I know, at-spi is just an accessibility daemon, which I don't have

**Reply by duarm** [reply]
I've also can't render helvum's node view, so it's not just context menus
![image](https://github.com/wwmm/easyeffects/assets/26150581/6ab60dde-940f-47ae-be12-e852f590a09a)

and I have compositor running, tried both picom and xcompmgr and without any

**Reply by wwmm** [reply]
>same problem on gtk4-demos, the logs only show this

What is your GPU? Are you using a Mesa package compiled with `LTO`? For some reason gtk4 apps are not rendered properly when Mesa is built with LTO support. Some weeks ago I opened an issue about this in the Mesa site https://gitlab.freedesktop.org/mesa/mesa/-/issues/10498 and one of its developers said that LTO is unfortunately not supported when compiling Mesa because of random bugs it can introduce like making apps crash or being incorrectly rendered.

From time to time Mesa's package in the Arch repositories is built with LTO enabled. So this may be your problem.

**Reply by duarm** [reply]
it's nvidia and I'm using mesa indeed, it worked before an update I did 2~3 months ago, so it is something like that most likely, thanks for taking your time to dig that up

**Reply by wwmm** [reply]
>it's nvidia and I'm using mesa indeed, it worked before an update I did 2~3 months ago, so it is something like that most likely, thanks for taking your time to dig that up

Hum... If it is using Mesa's nouveau driver instead of the proprietary nvidia driver then the cause is probably the repository package being built with LTO. But if it is the proprietary driver the one actualy handling the card then something different is happening.

**Comment by duarm** [comment]
Updating gtk4 to 4.14.1 fixed the problem, it just got released in arch's main repositories
